promql/functions.go

```diff
@@ -13,6 +13,18 @@
 
 package promql
 
+import (
+	"container/heap"
+	"math"
+	"sort"
+	"strconv"
+	"time"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+
+	"github.com/prometheus/prometheus/storage/metric"
+)
+
 // Function represents a function of the expression language and is
 // used by function nodes.
 type Function struct {
```

Comment:
As commented below, we are also designing packages (c.f. again: the PR to make `prepare*` exported). So as long as there is no good reason to forbid access, I think this is a fundamental part of our QL and should be available.

Even for us, who can export stuff when we need it, this is a valid concern. The AST being completely unexported was a major reason for making a new one along with the parser instead of changing hundreds of lines of code.